skimage/typing/_image_types.py

```diff
@@ -0,0 +1,14 @@
+from ._array_like import ArrayLike
+import numpy as np
+
+from typing_extensions import Annotated
+
+ImageArray = Annotated[np.ndarray, 'image',
+                        'a 2D or 3D single or multichannel image']
+Image2dArray = Annotated[np.ndarray, 'image_2d']
+ImagendArray = Annotated[np.ndarray, 'image_nd',
+                            'nd image']
```

Comment:
I'd strongly prefer for `Image` to be nD, and  `Image2D` and `Image3D` to be special cases.

Reply:
:+1:

Reply:
Also, `Array` may be unnecessary.

Reply:
Most often we want to specify either "N-d" or "2/3D".

Reply:
So we have
- N-d
- 2/3D 
- 2D only (for example `segmentation.active_contour` or `segmentation.chan_vese`)

Should we pick up Image3d for 2/3d or will it be confusing to people? Looking for a good name here.

Reply:
We also have multichannel vs spatial to consider.

Maybe `Image2D3D`? Your suggestion of `Image3D` for both can also work.

Reply:
Looking at the above again, having `Image3D` double for both would probably be fine?

Reply:
Are you aware of the [NewType](https://docs.python.org/3/library/typing.html#newtype) feature? On first glance it looks like this is what you actually want to use. `Annotated` is _ignored_ by mypy itself and supposed to be used for third party extensions to mypy. `NewType` ensures that only something that is specifically typed as `Image` is passed and not just any ndarray.